sklearn/ensemble/_forest.py

```diff
@@ -345,6 +345,14 @@ def fit(self, X, y, sample_weight=None):
 
         # Check parameters
         self._validate_estimator()
+        # TODO: Remove in v1.2
+        if self.criterion == "mse":
```

Comment:
This is also "hacky" OOP code, but it is more explicit about when we are raising this error.

```suggestion
        if isinstance(self, (RandomForestRegressor, ExtraTreesRegressor)) and self.criterion == "mse":
```